op-plasma/daclient.go

```diff
@@ -0,0 +1,93 @@
+package plasma
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// ErrNotFound is returned when the server could not find the input.
+var ErrNotFound = errors.New("not found")
+
+// ErrCommitmentMismatch is returned when the server returns the wrong input for the given commitment.
+var ErrCommitmentMismatch = errors.New("commitment mismatch")
+
+// ErrInvalidInput is returned when the input is not valid for posting to the DA storage.
+var ErrInvalidInput = errors.New("invalid input")
+
+// DAClient is an HTTP client to communicate with a DA storage service.
+// It creates commitments and retrieves input data + verifies if needed.
+// Currently only supports Keccak256 commitments but may be extended eventually.
+type DAClient struct {
+	url    string
+	verify bool
+}
+
+func NewDAClient(url string) *DAClient {
+	return &DAClient{url: url}
+}
+
+// VerifyOnRead sets the client to verify the commitment on read.
+// SHOULD enable if the storage service is not trusted.
+func (c *DAClient) VerifyOnRead(verify bool) {
+	c.verify = verify
+}
+
+// GetInput returns the input data for the given commitment bytes.
+func (c *DAClient) GetInput(ctx context.Context, key []byte) ([]byte, error) {
+	k := hexutil.Bytes(key)
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, fmt.Sprintf("%s/get/%s", c.url, k), nil)
```

Comment:
If we are using `fmt.Sprintf`, you can use `%x` or `%X` to get hex (lowercase or uppercase respectively)